New Resource: `azurerm_system_center_virtual_machine_manager_server` (#24278)
hashicorp/terraform-provider-azurerm: website/docs/r/system_center_virtual_machine_manager_server.html.markdown

---
subcategory: "System Center Virtual Machine Manager"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_system_center_virtual_machine_manager_server"
description: |-
  Manages a System Center Virtual Machine Manager Server.
---

# azurerm_system_center_virtual_machine_manager_server

Manages a System Center Virtual Machine Manager Server.

~> **Note:** By request of the service team the provider no longer automatically registering the `Microsoft.ScVmm` Resource Provider for this resource. To register it you can run `az provider register --namespace Microsoft.ScVmm`.

> 

Reviewer: Since this resource depends on some dependencies like an existing SCVMM environment and [Azure Arc resource bridge](https://learn.microsoft.com/en-us/azure/azure-arc/resource-bridge/overview), etc. Could you add a similar `note` like what [azurerm_arc_kubernetes_cluster](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/website/docs/r/arc_kubernetes_cluster.html.markdown#azurerm_arc_kubernetes_cluster) does to illustrate that the deployment of these dependencies is outside scope of this document and meanwhile provide some MS official guidance links of how to deploy these resources?

Author: I will confirm with service team about the related MS official guidance links.

---
subcategory: "System Center Virtual Machine Manager"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_system_center_virtual_machine_manager_server"
description: |-
  Manages a System Center Virtual Machine Manager Server.
---

# azurerm_system_center_virtual_machine_manager_server

Manages a System Center Virtual Machine Manager Server.

~> **Note:** By request of the service team the provider no longer automatically registering the `Microsoft.ScVmm` Resource Provider for this resource. To register it you can run `az provider register --namespace Microsoft.ScVmm`.

> -> **Note:** This resource depends on an existing `System Center Virtual Machine Manager Host Machine`, `Arc Resource Bridge` and `Custom Location`. Installing and configuring these dependencies is outside the scope of this document. See [Virtual Machine Manager documentation](https://learn.microsoft.com/en-us/system-center/vmm/?view=sc-vmm-2022) and [Install VMM](https://learn.microsoft.com/en-us/system-center/vmm/install?view=sc-vmm-2022) for more details of `System Center Virtual Machine Manager Host Machine`. See [What is Azure Arc resource bridge](https://learn.microsoft.com/en-us/azure/azure-arc/resource-bridge/overview) and [Overview of Arc-enabled System Center Virtual Machine Manager](https://learn.microsoft.com/en-us/azure/azure-arc/system-center-virtual-machine-manager/overview) for more details of `Arc Resource Bridge/Appliance`. See [Create and manage custom locations on Azure Arc-enabled Kubernetes](https://learn.microsoft.com/en-us/azure/azure-arc/kubernetes/custom-locations) for more details of `Custom Location`. If you encounter issues while configuring, we'd recommend opening a ticket with Microsoft Support.

Reviewer: According to this [Architecture figure](https://learn.microsoft.com/en-us/azure/azure-arc/system-center-virtual-machine-manager/overview#architecture), this resource depends on a connected `Arc Resource Bridge` and `SCVMM Management Server`. So the first sentence should not contain `Custom Location` ?

Author: It's [required property](https://github.com/Azure/azure-rest-api-specs/blob/d1eb3c20113d1018f25a8d97fdfa5f8bb5c659ea/specification/scvmm/resource-manager/Microsoft.ScVmm/stable/2023-10-07/scvmm.json#L2924) so that we need to call it out in tf doc.

description: |-
  Manages a System Center Virtual Machine Manager Server.
---

# azurerm_system_center_virtual_machine_manager_server

Manages a System Center Virtual Machine Manager Server.

~> **Note:** By request of the service team the provider no longer automatically registering the `Microsoft.ScVmm` Resource Provider for this resource. To register it you can run `az provider register --namespace Microsoft.ScVmm`.

## Example Usage

```hcl
resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

resource "azurerm_system_center_virtual_machine_manager_server" "example" {
  name                = "example-scvmmms"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location
  custom_location_id  = "/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resGroup1/providers/Microsoft.ExtendedLocation/customLocations/customLocation1"
  fqdn                = "example.labtest"

  credential {
    username = "testUser"
    password = "H@Sh1CoR3!"
  }
}
```

## Arguments Reference

The following arguments are supported:

> * `name` - (Required) The name which should be used for this System Center Virtual Machine Manager Server. Changing this forces a new resource to be created.

Reviewer: ```suggestion
* `name` - (Required) The name of the System Center Virtual Machine Manager Server. Changing this forces a new resource to be created.
```

Author: Updated

* `resource_group_name` - (Required) The name of the Resource Group where the System Center Virtual Machine Manager should exist. Changing this forces a new resource to be created.

* `location` - (Required) The Azure Region where the System Center Virtual Machine Manager Server should exist. Changing this forces a new resource to be created.

* `credential` - (Required) A `credential` block as defined below.

* `custom_location_id` - (Required) The ID of the Custom Location for the System Center Virtual Machine Manager Server. Changing this forces a new resource to be created.

* `fqdn` - (Required) The FQDN of the System Center Virtual Machine Manager Server. Changing this forces a new resource to be created.

> * `port` - (Optional) The port that is used to listened by the System Center Virtual Machine Manager Server. Possible values are between `1` and `65535`. Changing this forces a new resource to be created.

Reviewer: ```suggestion
* `port` - (Optional) The port on which the System Center Virtual Machine Manager Server is listening. Possible values are between `1` and `65535`. Changing this forces a new resource to be created.
```

Author: Updated

~> **Note:** By request of the service team the provider no longer automatically registering the `Microsoft.ScVmm` Resource Provider for this resource. To register it you can run `az provider register --namespace Microsoft.ScVmm`.

> -> **Note:** The dependencies `System Center Virtual Machine Manager Host Machine`, `Arc Resource Bridge/Appliance` and `Custom Location` require additional configuration. For `System Center Virtual Machine Manager Host Machine`, see more details from [Virtual Machine Manager documentation](https://learn.microsoft.com/en-us/system-center/vmm/?view=sc-vmm-2022) and [Install VMM](https://learn.microsoft.com/en-us/system-center/vmm/install?view=sc-vmm-2022). For `Arc Resource Bridge/Appliance`, see more details from [What is Azure Arc resource bridge](https://learn.microsoft.com/en-us/azure/azure-arc/resource-bridge/overview) and [Overview of Arc-enabled System Center Virtual Machine Manager](https://learn.microsoft.com/en-us/azure/azure-arc/system-center-virtual-machine-manager/overview). For `Custom Location`, see more details from [Create and manage custom locations on Azure Arc-enabled Kubernetes](https://learn.microsoft.com/en-us/azure/azure-arc/kubernetes/custom-locations). If you encounter issues while configuring, we'd recommend opening a ticket with Microsoft Support.

Reviewer: ```suggestion
-> **Note:** This resource depends on an existing `System Center Virtual Machine Manager Host Machine` and `Arc Resource Bridge`. Installing and configuring these dependencies is outside the scope of this document. See [Virtual Machine Manager documentation](https://learn.microsoft.com/en-us/system-center/vmm/?view=sc-vmm-2022) and [Install VMM](https://learn.microsoft.com/en-us/system-center/vmm/install?view=sc-vmm-2022) for more details of `System Center Virtual Machine Manager Host Machine`. See [What is Azure Arc resource bridge](https://learn.microsoft.com/en-us/azure/azure-arc/resource-bridge/overview) and [Overview of Arc-enabled System Center Virtual Machine Manager](https://learn.microsoft.com/en-us/azure/azure-arc/system-center-virtual-machine-manager/overview) for more details of `Arc Resource Bridge/Appliance`. See [Create and manage custom locations on Azure Arc-enabled Kubernetes](https://learn.microsoft.com/en-us/azure/azure-arc/kubernetes/custom-locations) for more details of `Custom Location`. If you encounter issues while configuring, we'd recommend opening a ticket with Microsoft Support.
```

Author: Updated
